### PR TITLE
Hs 1 migrate to ts

### PR DESCRIPTION
**Pull request by derlagg**

### CODE REVIEW

File: src/scripts/ci/publish.ts [hunk 1]
```diff
@@ -65,70 +67,73 @@ if (isSnapshotVersion) {
...
+function setVersionToDistPackage(packageVersion: string): void {
+    console.log('update version in dist package.json to:', packageVersion);
 
-    const packageJson = readJsonFile(`${dist}/package.json`);
-    packageJson.version = version;
+    const distPackageJsonFile = readJsonFile(`${dist}/package.json`);
+    distPackageJsonFile.version = packageVersion;
 
-    const packageJsonStringified = JSON.stringify(packageJson, undefined, 4);
+    const indent = 4;
```
Comment (tohasan) [inline src/scripts/ci/publish.ts]: Maybe it is better to create separate function instead of duplicating.

File: src/scripts/ci/publish.ts [hunk 2]
```diff
@@ -65,70 +67,73 @@ if (isSnapshotVersion) {
...
 
-function merge(branchName) {
-    run(`git merge --no-ff ${branchName}`)
+function merge(branchName: string): void {
+    run(`git merge --no-ff ${branchName}`);
 }
 
-function getCurrentBranchName() {
-    const branchList = run('git branch');
+function getCurrentBranchName(): string| undefined {
```
Comment (tohasan) [inline src/scripts/ci/publish.ts]: We seem to always get string as return value.

File: src/scripts/project/remove-files-by-pattern.ts [hunk 3]
```diff
@@ -3,18 +3,20 @@
  *
  * Parameters:
  *  --searchDir Directory where script will do search
+ *
```
Comment (tohasan) [inline src/scripts/project/remove-files-by-pattern.ts]: There is a strange thing with `@type {string}`. Maybe we should remove such line and add type near parameter definition.

```
@param {string} searchDir Directory where script will do search
@param {string} pattern RegExp pattern without slashes (/). For example, styles.d.ts$ becomes /styles.d.ts$/

@example
    
    bundle-dts.ts --searchDir ./some/dir --pattern styles.d.ts$
```

Link to jsdoc cheetsheet: https://devhints.io/jsdoc

File: src/scripts/common/args-parser/args-parser.spec.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { ArgsParser } from './args-parser';
+
+describe('ArgsParser', () => {
+
+    it('should correct parse empty args', () => {
```
Comment (tohasan) [inline src/scripts/common/args-parser/args-parser.spec.ts]: We usually add `describe` for method we check: `describe('#get', ...)`.

File: .travis.yml [hunk 5]
```diff
@@ -11,9 +11,9 @@ if: branch = master
 script:
   - git checkout $TRAVIS_BRANCH
   - git remote set-url origin https://sergeevNick:$GITHUB_API_KEY@github.com/eigen-space/helper-scripts.git
-  - chmod +x ./scripts/ci/detect-autocommit.sh
-  - if ./scripts/ci/detect-autocommit.sh ; then
-      npm run deploy:publish;
+  - chmod +x ./src/scripts/ci/detect-autocommit.sh
```
Comment (tohasan) [inline .travis.yml]: Maybe it is better to move this script out of directory `src`.

File: package.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
     "name": "@eigenspace/helper-scripts",
     "version": "1.0.73",
-    "main": "index.js",
+    "main": "src/index.ts",
```
Comment (tohasan) [inline package.json]: We also should not forget about built package where there are not typescript files.

File: src/scripts/common/args-parser/args-parser.spec.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+        expect(ArgsParser.get([])).toEqual(new Map([['_', null]]));
+    });
+
+    it('should correct parse empty values args', () => {
+        expect(ArgsParser.get(['-name'])).toEqual(new Map([['_', null], ['name', null]]));
+    });
+
+    it('should parse args without key', () => {
+        expect(ArgsParser.get(['cat', 'dog'])).toEqual(new Map([['_', ['cat', 'dog']]]));
+        expect(ArgsParser.get(['cat', '-animal', 'dog'])).toEqual(new Map([['_', ['cat']], ['animal', ['dog']]]));
```
Comment (tohasan) [inline src/scripts/common/args-parser/args-parser.spec.ts]: I suggest to split this case into separate spec.

File: src/scripts/common/args-parser/args-parser.spec.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    });
+
+    it('should parse keys starting with a hyphen', () => {
+        expect(ArgsParser.get(['-animal', 'dog'])).toEqual(new Map([['_', null], ['animal', ['dog']]]));
+        expect(ArgsParser.get(['--animal', 'dog'])).toEqual(new Map([['_', null], ['animal', ['dog']]]));
+        expect(ArgsParser.get(['ani-mal', 'dog'])).toEqual(new Map([['_', ['ani-mal', 'dog']]]));
```
Comment (tohasan) [inline src/scripts/common/args-parser/args-parser.spec.ts]: This case is not related to the spec name.

**Comment by derlagg** [comment]
Recreate pull request based on comments on changes
